app/models/manageiq/providers/redhat/infra_manager/api_integration.rb

```diff
@@ -273,9 +273,15 @@ def raw_connect_v4(server, port, path, username, password, service, scheme = 'ht
       # Get the timeout from the configuration:
       timeout, = ems_timeouts(:ems_redhat, service)
 
-      # Create the connection:
+      url = URI::Generic.build(
+        :scheme => scheme,
+        :host   => server,
+        :port   => port,
+        :path   => path
+      )
+
       OvirtSDK4::Connection.new(
-        :url      => "#{scheme}://#{server}:#{port}#{path}",
+        :url      => url.to_s,
```

Comment:
The oVirt SDK already accepts and handles correctly URLs with IPv6. Internally it uses `libcurl` to do all the HTTP work, and `libcurl` handlers URLs correctly, for all IPv6 versions.

However, I can change the SDK so that it also accepts URI objects, that is not a problem. But that would be in a future version, not in the one currently used by ManageIQ.